hasher-matcher-actioner/webapp/src/pages/settings/ThreatExchangeSettingsTab.jsx

```diff
@@ -141,7 +149,7 @@ export default function ThreatExchangeSettingsTab() {
       </Card.Header>
       <Card.Body>
         <Row className="mt-3">
-          <Spinner hidden={loading} animation="border" role="status">
+          <Spinner hidden={!loading} animation="border" role="status">
```

Comment:
I took the new version of this file and put it in my local environment and I'm still getting an error because there is a time when datasets is undefined, and so there won't be a .length property. It's the same error as this morning. I think you need `!datasets || datasets.length === 0` as the condition at line 155. But that then leads to another error at line 177, because again, there is a time when datasets is undefined, so `some` cannot be referenced.

I think you need to do a terraform desctroy then terraform apply to recreate the scenario I'm in.

Reply:
`datasets` is initialized to `[]` so I didn't think that would be an issue. Does a bad API request make it undefined? It feels like this check might just hide an API level issues.